internal/ingress/controller/controller.go

```diff
@@ -969,6 +991,54 @@ func (n *NGINXController) createServers(data []*extensions.Ingress,
 	return servers
 }
 
+// Compares an Ingress of a potential alternative backend's rules with each existing server and finds matching host + path pairs.
+// If a match is found, we know that this server should back the alternative backend and add the alternative backend
+// to a backend's alternative list.
+// If no match is found, then the serverless backend is deleted.
+func mergeAlternativeBackends(ing *extensions.Ingress, upstreams map[string]*ingress.Backend,
+	servers map[string]*ingress.Server) {
+
+	for _, rule := range ing.Spec.Rules {
+		for _, path := range rule.HTTP.Paths {
+			upsName := upstreamName(ing.Namespace, path.Backend.ServiceName, path.Backend.ServicePort)
+
+			ups := upstreams[upsName]
+
+			if ups.NoServer {
```

Comment:
Is this evey going to be `false`? Since we call this function iff the ingress is annotated with `canary`, the corresponding upstreams will all have `NoServer` set to `true`, right?

Reply:
There was a point before I refactored where it could have been been false, but that shouldn't be the case anymore. I can probably remove `NoServer` altogether